lib/Doctrine/DBAL/Query/QueryBuilder.php

```diff
@@ -465,6 +466,27 @@ public function select($select = null) : self
         return $this->add('select', $selects);
     }
 
+    /**
+     * Adds a DISTINCT flag to this query.
+     *
+     * <code>
+     *     $qb = $conn->createQueryBuilder()
+     *         ->select('u.id')
+     *         ->distinct()
+     *         ->from('users', 'u');
+     * </code>
+     *
+     * @param bool $flag If true, then DISTINCT is added to the query
+     *
+     * @return $this This QueryBuilder instance.
+     */
+    public function distinct(bool $flag = true)
+    {
+        $this->sqlParts['distinct'] = (bool) $flag;
```

Comment:
I don't think this cast is needed.
```suggestion
        $this->sqlParts['distinct'] = $flag;
```